aws/aws-parallelcluster
# 3.0: enhancements to API dockerfile and docs

* Converting OpenAPI specs to YAML format
* Minor improvements to API Dockerfile and build logic …
  * Upgrade pip to latest version
  * Change Docker build context to cli dir in order to reduce size and speed-up builds
  * Install API requirements as extras
  * Reorder instructions for better caching
* Add sample SAM template to locally test API Gateway integration
* Add API Readme file

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
